components/playlist/media_detector_component_manager.cc

```diff
@@ -255,6 +260,16 @@ bool MediaDetectorComponentManager::ShouldHideMediaSrcAPI(
                               });
 }
 
+const std::string& MediaDetectorComponentManager::GetMediaDetectorScript() {
+  if (!script_.empty())
+    return script_;
+
+  // In case we have yet to fetch the script, use local script instead. At the
+  // same time, fetch the script from component.
+  RegisterIfNeeded();
+  return GetLocalScript();
```

Comment:
@simonhong I think this is what you meant, isn't it?

Reply:
Oops, I commented before seeing the whole this PR :)